Add Spatial Continuity Prototype

paulhdk opened the pull request:
# Add Spatial Continuity Prototype

## :recycle: Current situation & Problem

Add the Spatial Continuity prototype.

## :gear: Release Notes
* Spatial Continuity visionOS app
* Spatial Continuity Camera iOS app

### Code of Conduct & Contributing Guidelines
By creating and submitting this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

paulhdk committed:
Add Spatial Continuity prototype

paulhdk committed:
Add CONTRIBUTORS.md

paulhdk committed:
Add README.md

paulhdk committed:
CI: update licencing comments

paulhdk committed:
CI: only run for iOS and visionOS

paulhdk committed:
fastlane: add WIP files

paulhdk committed:
CI: disable buildandtest for now

paulhdk committed:
Add missing licencing information

paulhdk committed:
licence: add MIT.txt

paulhdk committed:
swiftlint: add .swiftlint.yaml

paulhdk committed:
Localizable.xcstrings

paulhdk committed:
SC: SCC: address swiftlint warnings

paulhdk committed:
CI: disable periphery for now

paulhdk committed:
CI: disable CodeQL for now

paulhdk committed:
SCS: make CIImage extensions public

paulhdk committed:
GitHub: add code of conduct

paulhdk committed:
GitHub: add contribution guidelines

paulhdk committed:
GitHub: add security guidelines

paulhdk commented:
@PSchmiedmayer I have a few question re: the CI and was hoping you might have some pointers:
* How can we set up the CodeQL and fastlane to support multiple schemas? In my case, one for the visionOS app and one for the iOS app, i.e., two different apps, not the same app on multiple platforms.
* The periphery workflow doesn't appear to work because I have a Xcode project and not a Swift package project (?). Maybe we need to a add a mechanism here to pass the `--project` option? https://github.com/StanfordBDHG/.github/blob/main/.github/workflows/periphery.yml

PSchmiedmayer commented:
@paulhdk Thank you for continuing the work here!

- Fastlane & CodeQL: You should be able to modify the fast file to cover this track; I assume it is mainly about duplicating the work in the build action to build the one first, then the second one.
- Regarding periphery: We have that working in multiple apps, e.g. the PAWS project ([GitHub action](https://github.com/StanfordBDHG/PediatricAppleWatchStudy/blob/main/.github/workflows/static-analysis.yml#L31) + [configuration file](https://github.com/StanfordBDHG/PediatricAppleWatchStudy/blob/main/.periphery.yml))

paulhdk committed:
fastlane: build iOS app for testing

This enables the infrastructure required to run tests with fastlane.
Since no tests are present ATM, we only build for testing.

paulhdk committed:
codeql: enable for SCC

Currently not passing due to missing support for Swift 26?

paulhdk committed:
periphery: enable for SCC

paulhdk committed:
SCC: address periphery warnings

paulhdk committed:
periphery: exclude results from SpatialContinuityShared

paulhdk committed:
xcodeproj

paulhdk committed:
fastlane: codeql: enable Spatial Continuity app

paulhdk committed:
codeql: disable until current Swift toolchain is supported

paulhdk committed:
periphery: enable Spatial Continuity app

paulhdk committed:
periphery: explicitly add implicit arguments in .periphery.yml

paulhdk committed:
periphery: fix SpeziLLM exclusion in config

paulhdk committed:
periphery: excplicitly add targets of interest

paulhdk committed:
readme: list open source contributions

paulhdk commented:
> @paulhdk Thank you for continuing the work here!
> 
> * Fastlane & CodeQL: You should be able to modify the fast file to cover this track; I assume it is mainly about duplicating the work in the build action to build the one first, then the second one.
> 
> * Regarding periphery: We have that working in multiple apps, e.g. the PAWS project ([GitHub action](https://github.com/StanfordBDHG/PediatricAppleWatchStudy/blob/main/.github/workflows/static-analysis.yml#L31) + [configuration file](https://github.com/StanfordBDHG/PediatricAppleWatchStudy/blob/main/.periphery.yml))

Thank you! I believe I was able to get it running.
I've disabled CodeQL for now as it does not appear to support the current Swift toolchain yet. See https://github.com/github/codeql/pull/20446.

Do you know why the REUSE Compliance Check is showing up twice in the workflow runs?

PR is ready for review now!

PSchmiedmayer commented on CODE_OF_CONDUCT.md:
We have a shared one in our .github repo; we should link to that in the README + it also automatically shows up in the repos because it is there on an institutional level.

PSchmiedmayer commented on CONTRIBUTING.md:
Same as for the code of conduct; we have one on the institutional level.

PSchmiedmayer commented on LICENSE.md:
```diff
@@ -1,9 +1,21 @@
 MIT License
 
-Copyright (c) 2022 Stanford University
+Copyright (c) 2024, Paul Heidekrüger
```
We normally point to the contributors file; e.g. spezi: https://github.com/StanfordSpezi/Spezi/blob/main/LICENSE.md 

The file headers should generally say:
[`SPDX-FileCopyrightText: 2025 Stanford University and the project authors (see CONTRIBUTORS.md)`](https://github.com/StanfordSpezi/Spezi/blob/cdafc915979573e4b82ab1dc0eac5a0444222ed6/.swiftlint.yml#L4)

paulhdk replied:
What licence info should I add for the TUM logo(s)?

PSchmiedmayer replied:
I would make that one copyrighted by TUM, good point!

paulhdk replied:
And keep the MIT licence, too?

PSchmiedmayer replied:
Yes, the repo is still licensed under the license, different to the copyright.

PSchmiedmayer commented on README.md:
```diff
@@ -1,77 +1,60 @@
+* https://github.com/StanfordSpezi/SpeziAccount/pull/82
+* https://github.com/StanfordSpezi/SpeziLLM/pull/64
+* https://github.com/StanfordSpezi/SpeziViews/pull/43
+* https://github.com/StanfordSpezi/SpeziLLM/pull/61
```
We should add the following sections (contributing, license, our research, and footer image) at the bottom of the README like we have them in other repos, including the Biodesign Logo. You can also add a TUM logo next to it if you want to.

Example:
https://github.com/StanfordBDHG/PediatricAppleWatchStudy?tab=readme-ov-file#contributing

PSchmiedmayer commented on SECURITY.md:
Same as for the other files; we have them on an institutional basis.

PSchmiedmayer commented on README.md:
```diff
@@ -1,77 +1,60 @@
-- `APP_STORE_CONNECT_API_KEY_ID`: The key ID of the API key created in the App Store Connect API keys section.
-- `APP_STORE_CONNECT_API_KEY_BASE64`: The content of the key created in App Store Connect condensed into a Base64 representation, e.g., using `base64 -i AuthKey_ABCDEFGHIJ.p8 | pbcopy`.
+## Known Issues
 
-### Apple Xcode Certificate and Provisioning Profile
+* Connecting the Spatial Continuity and Spatial Continuity Camera apps may be unreliable at times. Suggestions and PRs welcome! 
```
I would rather document this as a GitHub issue; not a great place in the README to have this as a living document.

PSchmiedmayer commented on README.md:
```diff
@@ -1,77 +1,60 @@
 
-It first runs the Build and Test workflow to ensure all tests are passing.
-Once the Build and Test workflow passes, it builds the iOS application so it can be archived and sent to [TestFlight](https://developer.apple.com/testflight/) for internal beta deployment.
+## Functionality
```
It would be great to add some visuals here, either from the visionOS simulator or from the paper. 
Would be nice to have something visual here embedded in the README.

PSchmiedmayer reviewed (approved):
Nice; great job here @paulhdk!

I had a few minor comments on the setup; the code looks clean and well organized!

Happy to see this merged once the comments are addressed. I also fixed the REUSE GitHub action showing up twice and adjusted the required workflows. Totally fine to not have the CodeQL setup for now, let's hope they support Swift 6.2 soon ...

paulhdk committed:
github: use org-level CODE_OF_CONDUCT, CONTRIBUTING, and SECURITY files

paulhdk committed:
README: add "Our Research" Section

paulhdk committed:
license: update copyright info

paulhdk committed:
README: file 'Known Issues' as GitHub issues

paulhdk committed:
readme: use paths instead of links for TUM logos

paulhdk committed:
readme: update TUM logo sizes

paulhdk committed:
readme: render TUM logos through img tags

paulhdk committed:
xcodeproj

paulhdk committed:
readme: add screenshot

paulhdk commented:
Thanks for your comments, @PSchmiedmayer! I've addressed them all apart from the question above. PTAL.

PSchmiedmayer commented:
Thank you @paulhdk 🚀
Seems like the REUSE check also fails for the new image, once this and the TUM image is solved I can also merge the PR with admin permissions if there are no other things left to do 👍

paulhdk committed:
licence: add licence for SC_Prototype.png

paulhdk committed:
licence: fix incorrect or outdated licences

paulhdk committed:
licence: add licences for TUM logos

paulhdk committed:
licence: fix filenames for licences in .github/assets

paulhdk commented:
> Thank you @paulhdk 🚀 Seems like the REUSE check also fails for the new image, once this and the TUM image is solved I can also merge the PR with admin permissions if there are no other things left to do 👍

Done. Good to go from my side!

PSchmiedmayer committed:
Add Citation File

PSchmiedmayer committed:
Add Citation File
